api/src/main/java/org/openmrs/util/databasechange/UpdateCohortMemberIds.java

```diff
@@ -0,0 +1,109 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.util.databasechange;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import liquibase.change.custom.CustomTaskChange;
+import liquibase.database.Database;
+import liquibase.database.jvm.JdbcConnection;
+import liquibase.exception.CustomChangeException;
+import liquibase.exception.DatabaseException;
+import liquibase.exception.SetupException;
+import liquibase.exception.ValidationErrors;
+import liquibase.resource.ResourceAccessor;
+
+/**
+ * This change set is run to update cohort member ids
+ */
+public class UpdateCohortMemberIds implements CustomTaskChange {
+	
+	protected final static Log log = LogFactory.getLog(UpdateCohortMemberIds.class);
+	
+	/**
+	 * @see CustomTaskChange#execute(Database)
+	 */
+	@Override
+	public void execute(Database database) throws CustomChangeException {
+		JdbcConnection connection = (JdbcConnection) database.getConnection();
+		Statement stmt = null;
+		PreparedStatement pStmt = null;
+		
+		try {
+			stmt = connection.createStatement();
+			ResultSet rs = stmt
+					.executeQuery("SELECT @i:=0");
```

Comment:
I do not think this will work on PostgreSQL.
If it were as simple as this, i would not see a need for a custom java changeset. You would just use a liquibase sql tag. My thinking was that you were going to loop through the records in java while assigning an incrementing id. That way it would not be a database specific sql. Does it make sense? :)